File: cpp/8kyu/calculate-average.cpp
/*

Write a function which calculates the average of the numbers in a given array.

Note: Empty arrays should return 0.

*/

#include <vector>
using namespace std;

double calcAverage(const vector<int>& values){
    if(values.empty()) return 0;

    double sum =0;
    for(int val: values) {
        sum += val;
    }

    return sum / values.size();
}

File: java/8kyu/CalculateAverage.java
/*

Author: MARCEL EBALLAR
GitHub: https://github.com/MACLEB1903

Source:
https://www.codewars.com/kata/57a2013acf1fa5bfc4000921/train/java


Instructions:
Write a function which calculates the average of the numbers in a given array.
Note: Empty arrays should return 0.

*/

public class Kata{
  public static double find_average(int[] array){
    double sum = 0;
    for(int i=0; i<array.length; i++){
      sum+=array[i];
    }
    return sum/array.length;
  }
}